sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,50 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(*args):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if len(args) == 2:
+        n = args[0]
+        k = args[1]
+        if k > n:
+            return 0
+        if k >= n/2.0:
+            args = [n - k]
+    numparts = 0
+    if len(args) == 1:
+        n = args[0]
+        numparts = 1
+        p = [1]*(n + 1)
+        for i in range(1,n + 1):
+            numparts = 0
+            k = 1
+            l = 1
+            while 0 <= i-(l + k):
+                numparts = numparts - (-1)**k * (p[i - l] + p[i-(l + k)])
+                k = k + 1
+                l = l + 3 * k - 2
+
+            if 0 <= i - l:
+                numparts = numparts - (-1)**k * p[i - l]
+            p[i] = numparts
+
+    elif len(args) == 2:
+        n = args[0]
+        k = args[1]
+        if n == k or k == 1:
+            numparts = 1
+        elif n < k or k == 0:
+            numparts = 0
+        else:
+            q1 = int(n)
+            k1 = int(k)
+            p = [1]*q1
+            for i in range(2,k1 + 1):
+                for m  in range(i + 1,q1 - i + 1 + 1):
+                    p[m] = p[m] + p[m - i]
+                    numparts = p[q1 - k1 + 1]
```

Comment:
It looks like this is indented too far. It should not be necessary to assign repeatedly to `numparts` inside the double loop.